### DISCUSSION TITLE
HyperHDR V18 Worked over a year, upgraded to V19 never worked again

### DISCUSSION THREAD

**Discussion by strikeoutsandwich** [awawa-dev/HyperHDR#715] [Q&A] [answered]
I had a system using Raspberry Pi 3b+ and V18 that worked great for over a year (it used the ws281x driver and GPIO18 from the RPi).  For unknown reason, it stopped working one day.  I couldn't get it going again, so decided to upgrade to V19 along with improving my hardware solution to the following (my config files and recent log are both attached for reference)
[HyperHDR-19.0.0.0-Backup-2024-01-07.json](https://github.com/awawa-dev/HyperHDR/files/13854297/HyperHDR-19.0.0.0-Backup-2024-01-07.json)
[log.txt](https://github.com/awawa-dev/HyperHDR/files/13854298/log.txt)
[artifactSerialESP.zip](https://github.com/awawa-dev/HyperHDR/files/13854311/artifactSerialESP.zip);

-Raspberry Pi 3b+
-Alinan ESP32-S2FN4R2 ESP32-S2 (Lolin mini), flashed with custom firmware, attached for reference (factory version)
-Bridgold SN74AHCT125N SN74 level shift chip
-Hagibis USB 3.0 HDMI capture card
-Ezcoo 4k HDMI splitter
-Alitove 5v 20a power supply
-SK6812 RGBWW LED strip (total LEDs = 356)

I've spent the past several months troubleshooting (comparing to several similar cases in the discussions on this forum), to no avail.  I've verified my LEDs work with a separate Teensy OctoWS2811, and they worked fine.  Tried several sets of hardware and different cables.  These are the things I'm not sure of:

- It doesn't appear that I'm getting output from the Lolin mini.  Capture system shows input in the monitor.  When I try to get statistics using the "performance/log output" method shown at the bottom of the HyperSPI readme, all I get is a blank screen.
- I've tried both HyperSPI and HyperSerialESP32.  Currently my system is using HyperSerialSPI, in case my wiring is the issue, that one seemed simpler.
- It's hard to tell if the Lolin mini is running.  I have tried to find what the RST and "0" buttons do online, but haven't found a definitive description (at least that I understood).  Anyone know what those do, and is there a way to tell if the Lolin is running?
- Can you bypass the level shifter to see if it's the problem?  I didn't want to fry anything if it was taken out of the loop.

Any assistance would be greatly appreciated.

**Comment by awawa-dev** [comment, marked as answer]
Hi

The message after LED device is disabled/restarted confirms that there is a communication. You can perform few more test to confirm it.

> 2024-01-07T16:58:38.065Z [LEDDEVICE_ADALIGHT] Received: 'HyperHDR frames: 25 (FPS), receiv.: 25, good: 25, incompl.: 0, mem1: 3676, mem2: 3712, heap: 182512
RGBW => Gain: 255/255, red: 255, green: 255, blue: 255'

The ESP32-S2 seems to be working fine, why did you enable custom calibration if the LED doesn't work? It won't change anything, just turn it off for now. You can try bypassing the level shifter for now, but keep the data line as short as possible. The other most common problem with LEDs not working is common ground: a missing OR is used a very long ground connection or it has strange loops on it.

**Reply by strikeoutsandwich** [reply]
Thank you very much for your response.  I turned off the calibration, and re-checked the grounding loop.  Had ground and data pin swapped on my tester.  Moved the connector and viola!  It started working.  I feel a little dumb that that was the problem, but I'm sure I had that right on one of the other setups.  Oh well.  It's working now and that's all that matters.  Thanks again for pushing me in the right direction.
